main.js

```diff
@@ -0,0 +1,224 @@
+        <div class="mb-3">
+          <label for="exampleFormControlInput1" class="form-label">Name</label>
+          <input type="user" class="form-control commentName" placeholder="@username"/>
+        </div>
+        <div class="mb-3">
+          <label for="post" class="form-label">Post</label>
+          <textarea class="form-control commentText" rows="3" placeholder="This post is cool because..."></textarea>
+        </div>
+        <button type="button" class="btn btn-primary" id="post-btn">Add Comment</button>
+      </div>
+    </div>`;
+
+let posts = [
+  {
+    id: 1,
+    user: "My name",
+    post: "This is a post",
+    comments: [
+      {
+        id: 1,
+        comment: "This is a comment",
+        user: "User",
+      },
+    ],
+  },
+  {
+    id: 2,
+    user: "User 2",
+    post: "Post 2",
+    comments: [
+      {
+        id: 1,
+        comment: "This is a comment 2",
+        user: "Name",
+      },
+    ],
+  },
+];
+
+const stringToHTML = function (str) {
+  let div = document.createElement("div");
+  div.innerHTML = str;
+  div.className = "conatiner-fluid posts";
+  return div;
+};
+
+const uniqueID = function () {
+  const timestamp = Date.now();
+  const random = Math.floor(Math.random() * 10000);
+  const newId = `${timestamp}-${random}`;
+
+  const existingIds = [];
+  posts.forEach((post) => {
+    existingIds.push(post.id);
+    if (post.comments) {
+      post.comments.forEach((comment) => existingIds.push(comment.id));
+    }
+  });
+
+  if (existingIds.includes(newId)) {
+    return uniqueID();
+  }
+
+  return newId;
+};
+
+const preloadPosts = function () {
+  for (let i = 0; i < posts.length; i++) {
+    let commentsHTML = "";
+    if (posts[i].comments && posts[i].comments.length > 0) {
+      commentsHTML = `
+        <div class="comments-section" style="display: none;">
+        ${posts[i].comments
+          .map(
+            (comment) => `
+            <div class="comment" data-comment-id="${comment.id}">
+              <strong>${comment.user || "Anonymous"}:</strong>
+              <p>${comment.comment}</p>
+              <span class="deleteComment">delete</span>
+            </div>
+          `
+          )
+          .join("")}
+        </div>
+        `;
+    }
+
+    let postString = `
+      <h3>Posted by: ${posts[i].user}</h3>
+      <p>${posts[i].post}</p>
+      ${postStr}
+      ${commentsHTML}
+    `;
+    let postHTML = stringToHTML(postString);
+    postsFeed.prepend(postHTML);
+  }
+};
+
+const getPostInputs = function () {
+  submitBtn.addEventListener("click", function () {
+    posts.push({ id: uniqueID(), user: user.value, post: post.value });
+    let latestPost = posts.length - 1;
+    let postString = `
+    <h3>Posted by: ${posts[latestPost].user}</h3>
+    <p>${posts[latestPost].post}</p>
+    ${postStr}`;
+    let postHTML = stringToHTML(postString);
+    postsFeed.prepend(postHTML);
+  });
+};
+
+const handleComments = function () {
```

Comment:
this function is too big and its handling too much, you can extract some responsibilities to other functions like toggling visibility, adding comments, deleting comments. This will be very hard to debug or maintain and scale. Whenever you see such a big function in your code, that should turn a red or yellow flag.